demos/instr.h

```diff
@@ -157,6 +158,30 @@ int ExchangeFS(int input);
 // Returns the original value of ES and sets the new value.
 int ExchangeES(int input);
 
+// Performs a bound check with the bound instruction. Works only on 32-bit x86.
+// Must be inlined in order to avoid mispredicted jumps over it.
+SAFESIDE_ALWAYS_INLINE
+inline void BoundsCheck(const char *str, size_t offset) {
+  struct {
+    int32_t low;
+    int32_t high;
+  } string_bounds;
+
+  string_bounds.low = 0;
+  string_bounds.high = strlen(str);
+
+  // ICC and older versions of Clang have a bug in compiling the bound
+  // instruction. They swap the operands when translating C++ to assembler
+  // (basically changing the GNU syntax to Intel syntax) and afterwards the
+  // assembler naturally crashes with:
+  // Error: operand size mismatch for `bound'
+  // Therefore we have to hardcode the opcode to mitigate the ICC bug.
+  // The following line is the same as:
+  // asm volatile("bound %%rax, (%%rdx)"
+  //              ::"a"(offset), "d"(&string_bounds):"memory");
+  asm volatile(".word 0x0262"::"a"(offset), "d"(&string_bounds):"memory");
```

Comment:
mild preference for `.byte 0x62,0x02` to avoid having to think about endianness

Reply:
Done.